raiden-ts/src/utils/types.ts

```diff
@@ -276,3 +276,39 @@ export const instanceOf: <C extends Newable>(C: C) => t.Type<InstanceType<C>> =
       t.identity,
     ),
 );
+
+/**
+ * Infer type of last element of a tuple or array
+ * Currently supports tuples of up to 9 elements before falling back to array's inference
+ *
+ * FIXME: may be simplified with variadic tuple types from TypeScript 4.0
+ */
+export type Last<T extends any[]> = T extends [any, any, any, any, any, any, any, any, infer L]
+  ? L
+  : T extends [any, any, any, any, any, any, any, infer L]
+  ? L
+  : T extends [any, any, any, any, any, any, infer L]
+  ? L
+  : T extends [any, any, any, any, any, infer L]
+  ? L
+  : T extends [any, any, any, any, infer L]
+  ? L
+  : T extends [any, any, any, infer L]
+  ? L
+  : T extends [any, any, infer L]
+  ? L
+  : T extends [any, infer L]
+  ? L
+  : T extends [infer L]
+  ? L
+  : T[number] | undefined;
+
+/**
+ * Like lodash's last, but properly infer return type when argument is a tuple
+ *
+ * @param arr - Tuple or array to get last element from
+ * @returns Last element from arr
+ */
+export function last<T extends any[]>(arr: T): Last<T> {
```

Comment:
Thanks for introducing that. Was first about to complain while reviewing the former commit :grimacing: